### PR TITLE
added modulecalls

### LINKED ISSUE #9: Introduce Modulecalls
### What problem are you facing?

I am unable to implement my PR [#367](https://github.com/terraform-docs/terraform-docs/pull/367) on terraform-docs

### How could terraform-docs help solve your problem?

Can you please introduce Modulecalls into print settings and terraform module.

### PR DESCRIPTION
**Pull request by HeikoNeblung**
Signed-off-by: Heiko Neblung <Heiko.Neblung@t-systems.com>

### Description of your changes

Added support for modulecalls. 

Fixes #9 

I have:

- [x] Read and followed terraform-docs' [contribution process].
- [x] All tests pass when I run `make test`.

### How has this code been tested

[contribution process]: https://git.io/Jt3Hr

### CODE REVIEW

File: terraform/modulecall.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package terraform
```
Comment (khos2ow) [inline terraform/modulecall.go:1]: Please add the license header to this file.

File: terraform/modulecall.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+// Modulecall represents a submodule called by Terraform module.
+type Modulecall struct {
+	Name    string `json:"name"`
```
Comment (khos2ow) [inline terraform/modulecall.go:5]: Json tags are not really needed here, it should be safe to remove them.

File: terraform/module.go [hunk 3]
```diff
@@ -107,6 +118,7 @@ func WithOptionalInputs(oi []*Input) SectionFn {
 type module struct {
 	Header       string
 	Inputs       []*Input
+	Modulecalls  []*Modulecall
```
Comment (khos2ow) [inline terraform/module.go:121]: I think it should be better to use full camel case `ModuleCalls` rather than `Modulecalls`.

**Review by khos2ow — COMMENTED** [review]
Thank you @HeikoNeblung, I have some comments below otherwise LGTM.

**Comment by HeikoNeblung** [comment]
I've implemented your suggested changes. I will update [#367](https://github.com/terraform-docs/terraform-docs/pull/367) to match the changes here.

File: print/settings.go [hunk 4]
```diff
@@ -60,6 +60,12 @@ type Settings struct {
 	// scope: Global
 	ShowInputs bool
 
+	// ShowModulecalls show "ModuleCalls" information
+	//
+	// default: true
+	// scope: Global
+	ShowModulecalls bool
```
Comment (khos2ow) [inline print/settings.go:67]: Same comment for camelCase.

File: terraform/module.go [hunk 5]
```diff
@@ -62,6 +66,13 @@ func WithInputs(i []*Input) SectionFn {
 	}
 }
 
+// WithModulecalls adds Modulecalls to Module.
+func WithModulecalls(mc []*ModuleCall) SectionFn {
```
Comment (khos2ow) [inline terraform/module.go:70]: Same comment for camelCase.

File: terraform/module.go [hunk 6]
```diff
@@ -22,10 +22,11 @@ import (
 
 // Module represents a Terraform module. It consists of
 //
-// - Header       ('header' json key):    Module header found in shape of multi line comments at the beginning of 'main.tf'
-// - Inputs       ('inputs' json key):    List of input 'variables' extracted from the Terraform module .tf files
-// - Outputs      ('outputs' json key):   List of 'outputs' extracted from Terraform module .tf files
-// - Providers    ('providers' json key): List of 'providers' extracted from resources used in Terraform module
+// - Header       ('header' json key):        Module header found in shape of multi line comments at the beginning of 'main.tf'
+// - Inputs       ('inputs' json key):        List of input 'variables' extracted from the Terraform module .tf files
+// - Modulecalls  ('modulecalls' json key):   List of 'modulecalls' extracted from the Terraform module .tf files
```
Comment (khos2ow) [inline terraform/module.go:27]:
```suggestion
// - Modulecalls  ('modules' json key):   List of 'modules' extracted from the Terraform module .tf files
```

**Review by khos2ow — COMMENTED** [review]
Some small nit below!

**Review by khos2ow — APPROVED** [review]
Thank you @HeikoNeblung! LGTM, please rebase on `main` and we should be good to go. Also please squash your commits.

**Pull request merged**